site/src/components/Filter/filter.tsx

```diff
@@ -33,45 +34,75 @@ export type PresetFilter = {
 
 type FilterValues = Record<string, string | undefined>;
 
+type UseFilterConfig = {
+  /**
+   * The fallback value to use in the event that no filter params can be parsed
+   * from the search params object. This value is allowed to change on
+   * re-renders.
+   */
+  fallbackFilter?: string;
+  searchParamsResult: ReturnType<typeof useSearchParams>;
+  onUpdate?: (newValue: string) => void;
+};
+
+const useFilterParamsKey = "filter";
+
 export const useFilter = ({
-  initialValue = "",
-  onUpdate,
+  fallbackFilter = "",
   searchParamsResult,
-}: {
-  initialValue?: string;
-  searchParamsResult: ReturnType<typeof useSearchParams>;
-  onUpdate?: () => void;
-}) => {
+  onUpdate,
+}: UseFilterConfig) => {
   const [searchParams, setSearchParams] = searchParamsResult;
-  const query = searchParams.get("filter") ?? initialValue;
-  const values = parseFilterQuery(query);
-
-  const update = (values: string | FilterValues) => {
-    if (typeof values === "string") {
-      searchParams.set("filter", values);
-    } else {
-      searchParams.set("filter", stringifyFilter(values));
-    }
+  const query = searchParams.get(useFilterParamsKey) ?? fallbackFilter;
+
+  // Using ref approach to be defensive against useSearchParams not returning a
+  // stable set function on each render. Effect must run before query effect
+  const setSearchParamsRef = useRef(setSearchParams);
+  useEffect(() => {
+    setSearchParamsRef.current = setSearchParams;
+  }, [setSearchParams]);
+
+  // Keep params synced with query, even as query changes from outside sources
+  useEffect(() => {
+    const setSearchParams = setSearchParamsRef.current;
+
+    setSearchParams((currentParams) => {
+      const currentQuery = currentParams.get(useFilterParamsKey);
```

Comment:
`useFilterParamsKey` is not very clear to me what it is

Reply:
Maybe I could get rid of the `use` prefix in the name? Or maybe `urlParamsKey`?

The main goal with the variable is to make sure the same key is being used for all of `currentParams`'s reads/writes (it's already being used in four spots, so definitely don't want typos in the key name breaking things)